tls/s2n_server_cert_request.c

```diff
@@ -126,8 +151,21 @@ int s2n_client_cert_req_recv(struct s2n_connection *conn)
     return 0;
 }
 
+int s2n_tls13_cert_req_send(struct s2n_connection *conn)
+{
+    struct s2n_stuffer *out = &conn->handshake.io;
+
+    /* Write 0 length request context https://tools.ietf.org/html/rfc8446#section-4.3.2 */
+    GUARD(s2n_stuffer_write_uint8(out, 0));
+    /* write total extension length */
+    GUARD(s2n_stuffer_write_uint16(out, s2n_server_certificate_request_extensions_size(conn)));
+    /* write extensions */
+    GUARD(s2n_server_certificate_request_extensions_send(conn, out));
```

Comment:
Validation against this wire format would be easiest to check with an integ test. Until then, we're just checking against our interpretation of the rfc.

Reply:
The plan was to have an integ test with client auth as a whole and this is a subset of that. I can make an explicit note in the issue to have one before it can be closed, if you're alright with that?

Reply:
I made a note in the issue